assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -276,11 +316,46 @@ const baseResolvers = {
 			dispatch( STORE_NAME ).receiveAccountsPropertiesProfilesCompletion();
 		}
 
-		const accountID = registry.select( STORE_NAME ).getAccountID();
+		let accountID = registry.select( STORE_NAME ).getAccountID();
 		// Pre-select values from the matched property if no account is selected.
 		if ( matchedProperty && ! accountID ) {
-			registry.dispatch( STORE_NAME ).setAccountID( matchedProperty.accountId ); // eslint-disable-line sitekit/acronym-case
-			registry.dispatch( STORE_NAME ).selectProperty( matchedProperty.id, matchedProperty.internalWebPropertyId ); // eslint-disable-line sitekit/acronym-case
+			/* eslint-disable sitekit/acronym-case */
+			accountID = matchedProperty.accountId;
+			registry.dispatch( STORE_NAME ).setAccountID( matchedProperty.accountId );
+			yield Data.commonActions.await( registry.dispatch( STORE_NAME ).selectProperty( matchedProperty.id, matchedProperty.internalWebPropertyId ) );
+			/* eslint-enable */
+		}
+
+		// Bail out if the analytics-4 module is not enabled.
+		if ( ! isFeatureEnabled( 'ga4setup' ) ) {
+			return;
+		}
+
+		let ga4Property;
+		const ga4PropertyID = registry.select( MODULES_ANALYTICS_4 ).getPropertyID();
+		if ( ga4PropertyID ) {
+			ga4Property = yield Data.commonActions.await(
+				registry.__experimentalResolveSelect( MODULES_ANALYTICS_4 ).getProperty( ga4PropertyID )
+			);
+		}
+
+		// Bail out if the correct ga4 property is already selected.
+		if ( ga4Property?._accountID === accountID ) {
+			return;
+		}
+
+		yield Data.commonActions.await( registry.dispatch( MODULES_ANALYTICS_4 ).waitForProperties( accountID ) );
+
+		const ga4Properties = registry.select( MODULES_ANALYTICS_4 ).getProperties( accountID );
+		ga4Property = yield Data.commonActions.await(
+			registry.dispatch( MODULES_ANALYTICS_4 ).matchPropertyByURL(
+				ga4Properties.map( ( property ) => property._id ),
+				registry.select( CORE_SITE ).getReferenceSiteURL(),
+			)
+		);
+
+		if ( ga4Property?._id ) {
+			yield Data.commonActions.await( registry.dispatch( MODULES_ANALYTICS_4 ).selectProperty( ga4Property._id ) );
```

Comment:
This could use the same extracted action as mentioned above, the only difference I see is that this one doesn't set `GA4_PROPERTY_CREATE` if it isn't matched so maybe the `selectProperty` call wouldn't be included, or the action could take a "default" argument for the ID to select if there was no matched property.

Reply:
Updated.